### PR TITLE
MM-24595  Migrate API handler getChannelMember(s) to be compatible with local mode

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
Added local mode support to getChannelMember 
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-24595

### CODE REVIEW

File: api4/channel_test.go [hunk 1]
```diff
@@ -1634,40 +1634,47 @@ func TestGetChannelByNameForTeamName(t *testing.T) {
...
 	CheckUnauthorizedStatus(t, resp)
 
 	user := th.CreateUser()
-	Client.Login(user.Email, user.Password)
-	_, resp = Client.GetChannelMembers(th.BasicChannel.Id, 0, 60, "")
+	th.Client.Login(user.Email, user.Password)
+	_, resp = th.Client.GetChannelMembers(th.BasicChannel.Id, 0, 60, "")
 	CheckForbiddenStatus(t, resp)
 
 	_, resp = th.SystemAdminClient.GetChannelMembers(th.BasicChannel.Id, 0, 60, "")
```
Comment (mgdelacroix) [inline api4/channel_test.go:1680]: This test should be inside a `TestForSysadminAndLocal` helper

File: api4/channel_test.go [hunk 2]
```diff
@@ -1634,40 +1634,47 @@ func TestGetChannelByNameForTeamName(t *testing.T) {
...
 
-	_, resp = Client.GetChannelMembers("", 0, 60, "")
-	CheckBadRequestStatus(t, resp)
+		_, resp = client.GetChannelMembers("junk", 0, 60, "")
+		CheckBadRequestStatus(t, resp)
 
-	_, resp = Client.GetChannelMembers("junk", 0, 60, "")
-	CheckBadRequestStatus(t, resp)
+		_, resp = client.GetChannelMembers("", 0, 60, "")
+		// NOTE: for some reason, while using the LocalClient, the route /channels//members is not handled
```
Comment (mgdelacroix) [inline api4/channel_test.go:1659]: :eyes: can't figure out what happens here

File: api4/channel_test.go [hunk 3]
```diff
@@ -1716,29 +1723,29 @@ func TestGetChannelMember(t *testing.T) {
...
-	_, resp = Client.GetChannelMember("junk", th.BasicUser.Id, "")
-	CheckBadRequestStatus(t, resp)
+		_, resp = client.GetChannelMember("", th.BasicUser.Id, "")
+		CheckNotFoundStatus(t, resp)
 
-	_, resp = Client.GetChannelMember(model.NewId(), th.BasicUser.Id, "")
-	CheckForbiddenStatus(t, resp)
+		_, resp = client.GetChannelMember("junk", th.BasicUser.Id, "")
+		CheckBadRequestStatus(t, resp)
+		_, resp = Client.GetChannelMember(th.BasicChannel.Id, "", "")
```
Comment (mgdelacroix) [inline api4/channel_test.go:1737]:
```suggestion
		_, resp = client.GetChannelMember(th.BasicChannel.Id, "", "")
```

File: api4/channel_test.go [hunk 4]
```diff
@@ -1716,29 +1723,29 @@ func TestGetChannelMember(t *testing.T) {
...
+		CheckNotFoundStatus(t, resp)
 
-	_, resp = Client.GetChannelMember(th.BasicChannel.Id, "", "")
-	CheckNotFoundStatus(t, resp)
+		_, resp = Client.GetChannelMember(th.BasicChannel.Id, "junk", "")
```
Comment (mgdelacroix) [inline api4/channel_test.go:1740]:
```suggestion
		_, resp = client.GetChannelMember(th.BasicChannel.Id, "junk", "")
```

File: api4/channel_test.go [hunk 5]
```diff
@@ -1716,29 +1723,29 @@ func TestGetChannelMember(t *testing.T) {
 	th := Setup(t).InitBasic()
 	defer th.TearDown()
 	Client := th.Client
```
Comment (mgdelacroix) [inline api4/channel_test.go:1725]: For the sake of our sanity, I'd remove these assignments or change the name of the helper variable to `c` or something like that :stuck_out_tongue_closed_eyes: 

I wasn't expecting to find so many `Client := th.Client` around the tests when I wrote the proposal...

File: api4/channel_test.go [hunk 6]
```diff
@@ -1716,29 +1723,29 @@ func TestGetChannelMember(t *testing.T) {
...
+		CheckNotFoundStatus(t, resp)
+	})
 
-	_, resp = Client.GetChannelMember(th.BasicChannel.Id, model.NewId(), "")
-	CheckNotFoundStatus(t, resp)
+	_, resp := Client.GetChannelMember(model.NewId(), th.BasicUser.Id, "")
+	CheckForbiddenStatus(t, resp)
 
 	Client.Logout()
 	_, resp = Client.GetChannelMember(th.BasicChannel.Id, th.BasicUser.Id, "")
```
Comment (mgdelacroix) [inline api4/channel_test.go:1751]: Last test of the function (not the one in this line, but 1759) should be inside a `TestForSysadminAndLocal` helper

File: api4/channel_test.go [hunk 7]
```diff
@@ -1715,42 +1720,42 @@ func TestGetChannelMembersByIds(t *testing.T) {
...
 
 	user := th.CreateUser()
-	Client.Login(user.Email, user.Password)
-	_, resp = Client.GetChannelMember(th.BasicChannel.Id, th.BasicUser.Id, "")
+	c.Login(user.Email, user.Password)
+	_, resp = c.GetChannelMember(th.BasicChannel.Id, th.BasicUser.Id, "")
 	CheckForbiddenStatus(t, resp)
-
-	_, resp = th.SystemAdminClient.GetChannelMember(th.BasicChannel.Id, th.BasicUser.Id, "")
-	CheckNoError(t, resp)
```
Comment (isacikgoz) [inline api4/channel_test.go:1753]: We can add `th.TestForSystemAdminAndLocal` here :)

File: api4/channel_test.go [hunk 8]
```diff
@@ -1634,44 +1634,49 @@ func TestGetChannelByNameForTeamName(t *testing.T) {
...
 
 	user := th.CreateUser()
-	Client.Login(user.Email, user.Password)
-	_, resp = Client.GetChannelMembers(th.BasicChannel.Id, 0, 60, "")
+	th.Client.Login(user.Email, user.Password)
+	_, resp = th.Client.GetChannelMembers(th.BasicChannel.Id, 0, 60, "")
 	CheckForbiddenStatus(t, resp)
-
-	_, resp = th.SystemAdminClient.GetChannelMembers(th.BasicChannel.Id, 0, 60, "")
-	CheckNoError(t, resp)
```
Comment (isacikgoz) [inline api4/channel_test.go:1674]: Same as above.

**Pull request merged**